app/models/nomis/offender_base.rb

```diff
@@ -27,15 +27,23 @@ def convicted?
     def sentenced?
       # A prisoner will have had a sentence calculation and for our purposes
       # this means that they will either have a:
-      # 1) Release date, or
-      # 2) Parole eligibility date, or
-      # 3) HDC release date (homeDetentionCurfewEligibilityDate field).
-      # If they do not have any of these we should be checking for a tariff date
-      # Once we have all the dates we then need to display whichever is the
-      # earliest one.
+      # 1) Conditional release date (CRD) or
+      # 2) Automatic release date (ARD) or
+      # 3) Home detention curfew actual date (HDCAD) or
+      # 4) Post recall release date (PRRD) or
+      # 5) Sentence licence expiry date (SLED) or
+      # 6) Parole eligibility date (PED) or
+      # 7) Home detention curfew eligibility date (HDCED) or
+      # 8) Tariff end date (TED)
+      # 9) Or have an indeterminate sentence
+
       return false if sentence&.sentence_start_date.blank?
 
-      sentence.release_date.present? ||
+      sentence.conditional_release_date.present? ||
```

Comment:
The ticket suggested that we should check certain dates based on the sentence type. It would be really nice to do this, as it would start to put some real 'domain logic' into the codebase and start to make it a bit more encapsulated. It might be a pre-requirement that 'sentenceType' gets pushed into 'sentenceDetail' but that in itself sounds like a good thing. What do you think?

Reply:
Not sure the best way to do that

Reply:
I think there is a simpler solution than the one I proposed above - we can simply look at @sentence_type (e.g. @sentence_type.recall?) to decide which dates are valid in which scenarios